src/frame.js

```diff
@@ -0,0 +1,104 @@
+class FrameNotTen {
```

Comment:
could you use more private variables/functions in this class for encapsulation purposes?